server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/PushSubscriptionSet.scala

```diff
@@ -105,18 +110,29 @@ object VerificationCodeUpdate {
   }
 }
 
+object ExpiresUpdate {
+  def parse(jsValue: JsValue): Either[PatchUpdateValidationException, Update] = jsValue match {
+    case JsString(aString) => Right(ExpiresUpdate(UTCDate(ZonedDateTime.parse(aString))))
```

Comment:
Wrap `ZonedDateTime.parse(aString)` in a `Try` monad, convert it to an either and carry unexpected date parsing failures through the type system.

By the way, do we have a test for an invalid input as an expire? Today pretty sure we kinda get a serverFail :-p